recipe/patches/0001-manual-ucrt-libs.patch

```diff
@@ -0,0 +1,34 @@
+From 776876a9d9e6f094c3b3680e84734579e7a7f529 Mon Sep 17 00:00:00 2001
+From: Mervin Fansler <mervin.fansler@bric.ku.dk>
+Date: Thu, 5 Sep 2024 09:59:06 +0200
+Subject: [PATCH] manual ucrt libs
+
+---
+ src/Makevars.ucrt | 14 ++++----------
+ 1 file changed, 4 insertions(+), 10 deletions(-)
+
+diff --git a/src/Makevars.ucrt b/src/Makevars.ucrt
+index 8ff3402..fbbb02e 100644
+--- a/src/Makevars.ucrt
++++ b/src/Makevars.ucrt
+@@ -10,16 +10,10 @@ endif
+ 
+ PKG_CPPFLAGS = -I.
+ 
+-ifeq (,$(shell pkg-config --version 2>/dev/null))
+-  PKG_CPPFLAGS += -I$(LOCAL_SOFT)/include/cairo -I$(LOCAL_SOFT)/include/freetype2
+-  LIBSHARPYUV = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libsharpyuv.a),-lsharpyuv),)
+-  PKG_LIBS += -lcairo -lfontconfig -lintl -liconv -lexpat -lfreetype -lharfbuzz \
+-              -lpixman-1 -ltiff -lwebp $(LIBSHARPYUV) -llzma -lzstd -ljpeg -lpng -lbz2 \
+-              -lz -lgdi32 -lmsimg32
+-else
+-  PKG_CPPFLAGS += $(shell pkg-config --cflags cairo libtiff-4 libjpeg)
+-  PKG_LIBS += $(shell pkg-config --libs cairo libtiff-4 libjpeg)
+-endif
++PKG_LIBS += -L$(LIBRARY_LIB) -lcairo -lfontconfig -lintl -liconv -lexpat -lfreetype -lharfbuzz \
++            -lpixman-1 -ltiff -lwebp -llzma -lzstd -ljpeg -lpng -lbz2 \
++            -lz -lgdi32 -lmsimg32
++PKG_CPPFLAGS += -I$(LIBRARY_INC) -I$(LIBRARY_INC)/cairo -I$(LIBRARY_INC)/freetype2
```

Comment:
Could this be failing because the paths need to be separated by `\` instead of `/`?  So: 
```
-I$(LIBRARY_INC)\cairo -I$(LIBRARY_INC)\freetype2
```
instead of 
```
-I$(LIBRARY_INC)/cairo -I$(LIBRARY_INC)/freetype2
```
I'm not sure if R automatically fixes the path separators on Windows, but the compiler invocation seems to indicate it does not, at least when set from `PKG_CPPFLAGS`:
```
x86_64-w64-mingw32-gcc  -I"D:/bld/r-cairo_1725523740342/_h_env/lib/R/include" -DNDEBUG -I. -I%PREFIX%\Library\include -I%PREFIX%\Library\include/cairo -I%PREFIX%\Library\include/freetype2
```

Reply:
Hmm...I was thinking the opposite: The `(LIBRARY_INC) is coming out with `\` when it should be `/` in this context.

Reply:
Yes, now that I've seen more build output for Windows builds I think you are right.  The question now is how to replace `\` with `/`?  I'm not sure where `LIBRARY_INC` and `LIBRARY_LIB` are actually generated.

Reply:
Those are set by conda-build itself. I need to find my notes how to resolve this. The weird part is that I thought these work fine in other recipes in `Makevars.ucrt` (e.g., https://github.com/conda-forge/r-base-feedstock/issues/325#issuecomment-2267104279), but that perhaps because this one is a proper Makefile that they get evaluated in a different context?

I never seem to remember the tricks to this one 😮‍💨

Reply:
From https://www.gnu.org/software/make/manual/html_node/Text-Functions.html, perhaps:

```
LIBRARY_INC=$(subst \,/,$(LIBRARY_INC))
LIBRARY_LIB=$(subst \,/,$(LIBRARY_LIB))
```